File: trab3/ruby/secantes.rb
# encoding: utf-8
# Algoritmo SECANTES
require "./base"

Calcnum::algoritmo :secantes do |x0, x1|
  iterar do
    p = f(x1) / f(x0)
    x = x1 - (x0 - x1) * p / (1.0 - p)
    break x if (x - x1).abs < tol and f(x).abs < tol
    x0, x1 = x1, x
  end
end

File: trab3/python/secantes.py
from base import *


@algoritmo
def secantes(self, f, xi, xf):
    for _ in xrange(self.nmi):
        p = f(xf) / f(xi)
        x = xf - (xi - xf) * p / (1 - p)
        if abs(x - xf) < self.tol and abs(f(x)) < self.tol:
            return x
        xi, xf = xf, x
    raise ErroNMI

